### PR TITLE
test typescriptification - timeline-window, scheduler, etc

### PR DESCRIPTION
**Pull request by kerryarchibald**
https://github.com/matrix-org/matrix-js-sdk/issues/2353
Recommend viewing by commit

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/unit/timeline-window.spec.ts [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
-    if (baseIndex === undefined) {
-        baseIndex = 1;
-    }
-
-    // XXX: this is a horrid hack
-    const timelineSet = { room: { roomId: ROOM_ID } };
-    timelineSet.room.getUnfilteredTimelineSet = function() {
-        return timelineSet;
-    };
+function createTimeline(numEvents = 3, baseIndex = 1) {
```
Comment (robintown) [inline spec/unit/timeline-window.spec.ts:37]: This could use a return type

File: spec/unit/timeline-window.spec.ts [hunk 2]
```diff
@@ -38,7 +54,7 @@ function createTimeline(numEvents, baseIndex) {
 function addEventsToTimeline(timeline, numEvents, toStartOfTimeline) {
```
Comment (robintown) [inline spec/unit/timeline-window.spec.ts:54]: Untyped params

File: spec/unit/timeline-window.spec.ts [hunk 3]
```diff
@@ -59,6 +75,11 @@ function createLinkedTimelines() {
 }
```
Comment (robintown) [inline spec/unit/timeline-window.spec.ts:75]: Untyped return

File: spec/unit/timeline-window.spec.ts [hunk 4]
```diff
@@ -137,320 +158,288 @@ describe("TimelineWindow", function() {
...
-        return new TimelineWindow(client, timelineSet, opts);
+    function createWindow(timeline: EventTimeline, opts?: {
+        windowLimit?: number;
+    }): [TimelineWindow, EventTimelineSet] {
+        const timelineSet = { getTimelineForEvent: () => null } as unknown as EventTimelineSet;
+        // client = {};
+        // client.getEventTimeline = function(timelineSet0, eventId0) {
+        //     expect(timelineSet0).toBe(timelineSet);
+        //     return Promise.resolve(timeline);
+        // };
```
Comment (robintown) [inline spec/unit/timeline-window.spec.ts:169]: Can this be removed now?

**Review by robintown — APPROVED** [review]
All good otherwise

**Comment by kerryarchibald** [comment]
Sorry @robintown, didn't mean to clear your review. Thanks for reviewing!

**Review by robintown — APPROVED** [review]

**Pull request merged**